File: Kms.Security.Common/Domain/Organization.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Kms.Security.Common.Domain
{
   public class Organization: IDomainBase<Guid>
    {
        public Organization(Guid id, string name, Guid? parentId/*,string code,string address, string website,string phone*/, int sortingNumber)
        {
            Id = id;
            Name = name;
            ParentId = parentId;
            SortingNumber = sortingNumber;
            //Code = code;
            //Address = address;
            //Website = website;
            //Phone = phone;
        }
        protected Organization()
        {

        }

        public void SetNewId()
        {
            this.Id = Guid.NewGuid();
        }
        public bool IdIsEmpty()
        {
            return this.Id == Guid.Empty;
        }
        public void SetEnterprisePositions(IList<EnterprisePosition> enterprisePosins)
        {
            EnterprisePositions = enterprisePosins;
        }
        public IList<EnterprisePosition> EnterprisePositions { get; protected set; }
        public void AddParent(Organization parent)
        {
            this.Parent = parent;
        }
        public string Name { protected set; get; }

        public virtual Organization Parent { protected set; get; }

        public Guid? ParentId { protected set; get; }
        public Guid Id { get; protected set; }
        public int SortingNumber { get; set; }
        //public string Phone { get; protected set; }


        //public string Website { get; protected set; }

        //public string Address { get; protected set; }

        //public string Code { get; protected set; }

    }
}

File: Kms.Security.Portal/app/nodak/common/organization/organization.ts
module nodak.common.models {
    export class Organization extends nodak.models.ModelBase{
        constructor() {
            super(enums.SubSystems.Common, "Organization");
        }
        AddBussinessModelValidation() {
            
        }
        private id: string;
        set Id(value: string) {
            this.id = value;
        }
        get Id() {
            return this.id;
        }
        private name: string;
        set Name(value: string) {
            this.name = value;
        }
        get Name() {
            return this.name;
        }

        private parentId: string;
        set ParentId(value: string) {
            this.parentId = value;
        }
        get ParentId() {
            return this.parentId;
        }

        private parent:models.Organization;
        set Parent(value: Organization) {
            this.parent = value;
        }
        get Parent() {
            return this.parent;
        }

        private sortingNumber: number;
        set SortingNumber(value: number) {
            this.sortingNumber = value;
        }
        get SortingNumber() {
            return this.sortingNumber;
        }

        private enterprisePositions: any;
        set EnterprisePositions(value: string) {
            this.enterprisePositions = value;
        }
        get EnterprisePositions() {
            return this.enterprisePositions;
        }

        //private phone: string;
        //set Phone(value: string) {
        //    this.phone = value;
        //}
        //get Phone() {
        //    return this.phone;
        //}

        //private website: string;
        //set Website(value: string) {
        //    this.website = value;
        //}
        //get Website() {
        //    return this.website;
        //}

        //private address: string;
        //set Address(value: string) {
        //    this.address = value;
        //}
        //get Address() {
        //    return this.address;
        //}

        //private code: string;
        //set Code(value: string) {
        //    this.code = value;
        //}
        //get Code() {
        //    return this.code;
        //}


    }
}